File: server/data/results/translations/lite/anchored/cpp/balanced_parens.cpp
#include <iostream>
#include <string>
#include <vector>
#include <stack>
#include <unordered_map>

bool is_balanced(const std::string& text) {
    std::stack<char> stack;
    std::unordered_map<char, char> pairs = {{')', '('}, {']', '['}, {'}', '{'}};
    for (char ch : text) {
        if (ch == '(' || ch == '[' || ch == '{') {
            stack.push(ch);
        } else if (pairs.find(ch) != pairs.end()) {
            if (stack.empty()) {
                return false;
            }
            char top = stack.top();
            stack.pop();
            if (top != pairs[ch]) {
                return false;
            }
        }
    }
    return stack.empty();
}

void main_func() {
    std::vector<std::string> samples = {"()", "([{}])", "(]", "((()", "{[()]}="};
    for (const std::string& sample : samples) {
        if (is_balanced(sample)) {
            std::cout << "true" << std::endl;
        } else {
            std::cout << "false" << std::endl;
        }
    }
}

int main() {
    main_func();
    return 0;
}

File: server/data/results/translations/lite/anchored/javascript/balanced_parens.js
function is_balanced(text) {
    let stack = [];
    let pairs = {")": "(", "]": "[", "}": "{"};
    for (let i = 0; i < text.length; i++) {
        let ch = text[i];
        if (ch === "(" || ch === "[" || ch === "{") {
            stack.push(ch);
        } else if (ch in pairs) {
            if (stack.length === 0) {
                return false;
            }
            let top = stack.pop();
            if (top !== pairs[ch]) {
                return false;
            }
        }
    }
    return stack.length === 0;
}

function main() {
    let samples = ["()", "([{}])", "(]", "((()", "{[()]}"];
    for (let i = 0; i < samples.length; i++) {
        let sample = samples[i];
        if (is_balanced(sample)) {
            console.log("true");
        } else {
            console.log("false");
        }
    }
}

main();